File: Data Structures & Algorithms/minimum-remove-to-make-valid-parentheses/submission-0.py
class Solution:
    def minRemoveToMakeValid(self, s: str) -> str:
        
        i = 0
        s = list(s)
        stack = []

        while i < len(s):
            if s[i] == "(":
                stack.append(i)
            elif s[i] == ")" and stack:
                stack.pop()
            elif s[i] == ")" and not stack:
                s[i] = ''
    
            i = i + 1
        
        if stack:
            for i in stack:
                s[i] = ''

        return ''.join(s)

File: Data Structures & Algorithms/baseball-game/submission-0.js
class Solution {
    /**
     * @param {string[]} operations
     * @return {number}
     */
    calPoints(operations) {
        const ans = [];

        while (operations.length != 0) {
            const op = operations.shift();
            
            if (op == "D" && ans.length > 0) {
                ans.push(ans[ans.length - 1] * 2);
            } else if (op == "C" && ans.length > 0) {
                ans.pop();
            } else if (op == "+" && ans.length > 1) {
                const add1 = ans[ans.length - 1];
                const add2 = ans[ans.length - 2];
                ans.push(add1 + add2);
            } else {
                ans.push(parseInt(op));
            };
        };

        let output = 0;

        for (const an of ans) {
            output += an;
        };

        return output;
    };
};
